chemclipse/plugins/org.eclipse.chemclipse.msd.model/src/org/eclipse/chemclipse/msd/model/matrix/ExtractedMatrix.java

```diff
@@ -17,39 +17,42 @@
 import java.util.List;
 import java.util.stream.Collectors;
 
+import org.eclipse.chemclipse.msd.model.core.AbstractIon;
 import org.eclipse.chemclipse.msd.model.core.IIon;
 import org.eclipse.chemclipse.msd.model.core.IIonBounds;
 import org.eclipse.chemclipse.msd.model.core.IScanMSD;
 import org.eclipse.chemclipse.msd.model.core.selection.IChromatogramSelectionMSD;
 import org.eclipse.chemclipse.msd.model.implementation.Ion;
 
-public class ExtractedMatrix implements IExtractedMatrix {
+public class ExtractedMatrix {
 	
 	private IChromatogramSelectionMSD selection;
 	private List<IScanMSD> scans;
 	private int numberOfScans;
 	private int numberOfIons;
 	private int startIon;
 	private int stopIon;
-	private float[] signal;
-	private boolean highRes = false;
+	private float[][] signal;
 	
 	public ExtractedMatrix(IChromatogramSelectionMSD chromatogramSelection) {
 		this.selection = chromatogramSelection;
 		this.numberOfScans = selection.getStopScan() - selection.getStartScan() - 1;
 		this.scans = extractScans();
 		if(checkHighRes( 10)) {
-			highRes = true;
-			System.out.println("This is highres MS Data and should be binned first");
+			throw new IllegalArgumentException();
 		} else {
 			this.startIon = getMinMz();
 			this.stopIon = getMaxMz();
 			this.numberOfIons = this.stopIon - this.startIon + 1;
-			signal = new float[numberOfScans * (this.stopIon - this.startIon + 1)];
+			signal = new float[numberOfScans][this.numberOfIons];
+			List<IIon> currentIons;
+			int numberIonsCurrentScan;
 			try {
-				for(int i = 0; i < numberOfScans; i++) {
-					for(int j = 0; j < scans.get(i).getIons().size(); j++) {
-						signal[ ((int) Math.round(scans.get(i).getIons().get(j).getIon() - this.startIon)) * numberOfScans + i    ] =  scans.get(i).getIons().get(j).getAbundance();
+				for(int scanIndex = 0; scanIndex < numberOfScans; scanIndex++) {
+					currentIons = scans.get(scanIndex).getIons();
+					numberIonsCurrentScan = currentIons.size(); 
+					for(int j = 0; j < numberIonsCurrentScan; j++) {
+						signal[ ((int) Math.round(currentIons.get(j).getIon() - this.startIon))] [j] =  currentIons.get(j).getAbundance();
 					}
 				}
 			} catch (Exception e) {
```

Comment:
You should either throw exceptions here to the caller or catch and rethrow as a spcialized exception describing the cause.

Reply:
try/catch actually not even needed here. Will remove.